src/examples/combobox/NestedMultiselect.tsx

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { Col, Row } from '@zendeskgarden/react-grid';
+import {
+  Combobox,
+  Field,
+  Hint,
+  IComboboxProps,
+  IOptGroupProps,
+  IOptionProps,
+  Label,
+  OptGroup,
+  Option,
+  OptionValue
+} from '@zendeskgarden/react-dropdowns.next';
+
+interface IOption extends IOptionProps {
+  value: string;
+}
+interface IOptGroup extends IOptGroupProps {
+  options: IOption[];
+}
+
+type Options = (IOption | IOptGroup)[];
+
+type SubOptions = Record<any, Options>;
```

Comment:
[nit]

```suggestion
type SubOptions = Record<string, Options>;
```

Reply:
@ze-flo we'll get these type enhancements dialed-in for v9, where `OptionValue` is only valid for `string`. In order to let the `OptionValue = string | object` flow through the example, this has to be `any` 😞